Infrastructure.DataAccess/Mapping/ItContractMap.cs

```diff
@@ -58,6 +58,11 @@ public ItContractMap()
                 .WithMany(t => t.Children)
                 .HasForeignKey(d => d.ParentId)
                 .WillCascadeOnDelete(false);
+            
+            HasOptional(t => t.CriticalityType)
+                .WithMany(t => t.References)
+                .HasForeignKey(d => d.CriticalityId)
+                .WillCascadeOnDelete(false);
```

Comment:
the other choice type references do not have the additional "cascade" option... do you know if "false" is default.. if not, remove it.